ci/jenkins/utils.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function check_and_upgrade_golang() {
+    export GOROOT=/usr/local/go
+    export PATH=${GOROOT}/bin:$PATH
+    echo "====== Checking installed Golang version ======"
+    # Since golang release 1.21, the version format in go.mod is changed from "go 1.21" to "go1.21.0".
+    # But if the golang version on testbed is older than 1.21, command "go mod edit -json | jq -r '.Go'"
+    # will return an error. So we using the following command to get the golang version.
```

Comment:
I find this comment misleading:
1) **we should be able to install go even if there is no existing version of go on a machine**
2) we don't use `go mod edit -json` here, so maybe no need to mention it
3) `go-1.19 mod edit -json | jq -r '.Go'` works fine for me, so it doesn't seem to return an error

Reply:
Thanks for the reminder, I removed this comment. As for these questions, 
1. This script will install golang version if there's no golang installed on the machine, it just checks `/usr/local/go${version}` and link it to `/usr/local/go`.
2. Removed this command.
3. If your golang version is higher than go1.21, both `go1.20` and `go1.20.0` are acceptable, but if the golang version on the testbed is older than go1.21, format `go 1.21.0` will return an error. Just as follows, it will not install the newer version, that why I did not use `go mod edit` command. 
```bash
➜   go mod edit -json | jq -r '.Go'
go: errors parsing go.mod:
/root/antrea/go.mod:3: invalid go version '1.21.0': must match format 1.23
➜  go version
go version go1.20.4 linux/amd64
```